content/v2.0/introduction/getting-started.md

```diff
@@ -138,29 +162,1261 @@ Visit your application in the browser at [http://localhost:2300](http://localhos
+    Hanami.app.prepare(:persistence)
+    ROM::SQL::RakeSupport.env = ROM.container(Hanami.app["persistence.config"])
+  end
+end
 ```
 
-We'll see these commands at play in this guide.
 
-## Building your first application
+### Creating a books table
+
+With that set up out of the way (none of which will be necessary come Hanami 2.1!) we can now create a books table.
+
+To create a migration run `bundle exec rake db:create_migration[create_books]`:
+
+```shell
+bundle exec rake db:create_migration[create_books]
+<= migration file created db/migrate/20221113050928_create_books.rb
+```
+
+Edit the migration file in order to create a books table with title and author columns and a primary key:
+
+```ruby
+# db/migrate/20221113050928_create_books.rb
+
+# frozen_string_literal: true
+
+ROM::SQL.migration do
+  change do
+    create_table :books do
+      primary_key :id
+      column :title, :text, null: false
+      column :author, :text, null: false
+    end
+  end
+end
+```
+
+Migrate both the development and test databases:
+
+```shell
+bundle exec rake db:migrate
+HANAMI_ENV=test bundle exec rake db:migrate
+```
+
+Lastly, let's add a rom-rb relation to allow our application to interact with our books table. Create the following file at `lib/bookshelf/persistence/relations/books.rb`:
+
+```ruby
+# lib/bookshelf/persistence/relations/books.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  module Persistence
+    module Relations
+      class Books < ROM::Relation[:sql]
+        schema(:books, infer: true)
+      end
+    end
+  end
+end
+```
+
+## Listing books
+
+With our books table ready to use, let's adapt our books index spec to expect an index of persisted books:
+
+```ruby
+# frozen_string_literal: true
+
+RSpec.describe "GET /books", type: [:request, :database] do
+  let(:books) { app["persistence.rom"].relations[:books] }
+
+  before do
+    books.insert(title: "Refactoring", author: "Martin Fowler")
+    books.insert(title: "Test Driven Development", author: "Kent Beck")
+  end
+
+  it "returns a list of books" do
+    get "/books"
+
+    expect(last_response).to be_successful
+    expect(last_response.content_type).to eq("application/json; charset=utf-8")
+
+    response_body = JSON.parse(last_response.body)
+
+    expect(response_body).to eq([
+      { "title" => "Refactoring", "author" => "Martin Fowler" },
+      { "title" => "Test Driven Development", "author" => "Kent Beck" }
+    ])
+  end
+end
+```
+
+To get this spec to pass, we'll need to update our books index action to return books from the books relation.
+
+To access the books relation within the action, we can use Hanami's "Deps mixin". Covered in detail in the [containers and components](/v2.0/architecture/containers/) section of the [Architecture guide](/v2.0/architecture/containers/), the Deps mixin gives each of your application's components easy access to the other components it depends on to achieve its work. We'll see this in more detail in a moment.
```

Comment:
There are two links pointing to the same location. Maybe the first one should point to the exact section and the latter is not necessary?